[16.0][IMP] template_content_swapper: add feature to use domain for applying some specific records in report pdfs

AungKoKoLin1997 opened the pull request:
This PR adds an improvement to restrict the applying changes only for some specific records by adding domain in configuration.
With this changes, we can only adjust the content of report pdf depends on some condition, not globally.

@qrtl QT5806

AungKoKoLin1997 committed:
[IMP] template_content_swapper: add feature to use domain for applying some specific records in report pdfs

Co-authored-by: Yoshi Tashiro <tashiro@quartile.co>

yostashiro reviewed (approved):
@AungKoKoLin1997 I've refactored the code. Please review my commit and squash.

AungKoKoLin1997 commented:
> @AungKoKoLin1997 I've refactored the code. Please review my commit and squash.

@yostashiro Thanks for the refactoring code. 
Looks good and I squashed the commits.

nobuQuartile commented:
LGTM
Functional review and Code review

AungKoKoLin1997 commented:
In my last commit, I rework on the tests.

yostashiro commented on template_content_swapper/__manifest__.py:
```diff
@@ -1,9 +1,9 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 {
     "name": "Template Content Swapper",
```
Please add a summary as well.

AungKoKoLin1997 replied:
Done!

yostashiro commented on template_content_swapper/models/ir_qweb.py:
```diff
@@ -1,51 +1,118 @@
             return result
         result_str = str(result)
-        lang_code = "en_US"
         request = values.get("request")
```
As a precaution.
```suggestion
        values = values or {}
        request = values.get("request")
```

AungKoKoLin1997 replied:
Done!
